packages/frontend/core/src/desktop/pages/auth/confirm-change-email.tsx

```diff
@@ -0,0 +1,68 @@
+import { Button } from '@affine/component';
+import { AuthPageContainer } from '@affine/component/auth-components';
+import { useNavigateHelper } from '@affine/core/components/hooks/use-navigate-helper';
+import { BackendError, GraphQLService } from '@affine/core/modules/cloud';
+import {
+  changeEmailMutation,
+  ErrorNames,
+  UserFriendlyError,
+} from '@affine/graphql';
+import { useI18n } from '@affine/i18n';
+import { useService } from '@toeverything/infra';
+import { type FC, useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { AppContainer } from '../../components/app-container';
+
+export const ConfirmChangeEmail: FC<{
+  onOpenAffine: () => void;
+}> = ({ onOpenAffine }) => {
+  const t = useI18n();
+  const [searchParams] = useSearchParams();
+  const navigateHelper = useNavigateHelper();
+  const graphqlService = useService(GraphQLService);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      const token = searchParams.get('token') ?? '';
+      const email = decodeURIComponent(searchParams.get('email') ?? '');
+      await graphqlService
+        .gql({
+          query: changeEmailMutation,
+          variables: {
+            token: token,
+            email: email,
+          },
+        })
+        .catch(err => {
+          if (err instanceof BackendError) {
+            const userFriendlyError = UserFriendlyError.fromAnyError(err);
```

Comment:
originError in BackendError is UserFriendlyError already

Reply:
you can update the implementation of `fromAnyError` to extract it from BackendError

Reply:
if so, instanceof is not necessary